mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/client/DefaultOperationParameters.java

```diff
@@ -0,0 +1,110 @@
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.client;
+
+import static com.google.common.collect.ImmutableMap.copyOf;
+import static java.util.Optional.ofNullable;
+import org.mule.runtime.extension.api.client.OperationParameters;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import javafx.util.Pair;
+
+/**
+ * Default {@link OperationParameters} implementation, works as a generic {@link OperationParameters} for every extension operation.
+ * <p>
+ * This class is not part of the API and should not be used by anyone (or anything) but the runtime. Backwards compatibility
+ * not guaranteed on this class.
+ *
+ * @since 1.0
+ */
+public class DefaultOperationParameters implements OperationParameters {
+
+  private final String configRef;
+  private final Map<String, Object> parameters;
+  private final Map<String, Pair<Class, OperationParameters>> complexParams;
+
+  DefaultOperationParameters(String configRef,
+                             Map<String, Object> parameters,
+                             Map<String, Pair<Class, OperationParameters>> complexParams) {
+    this.configRef = configRef;
+    this.parameters = copyOf(parameters);
+    this.complexParams = copyOf(complexParams);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Optional<String> getConfigName() {
+    return ofNullable(configRef);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<String, Object> getParameters() {
+    return copyOf(parameters);
+  }
+
+  public Map<String, Pair<Class, OperationParameters>> getComplexParameters() {
+    return complexParams;
+  }
+
+  public static class Builder implements OperationParametersBuilder {
+
+    private String configRef;
+    private Map<String, Object> parameters = new LinkedHashMap<>();
+    private Map<String, Pair<Class, OperationParameters>> complexParams = new LinkedHashMap<>();
+
+    public static Builder getInstance() {
+      return new Builder();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Builder addParameter(String name, Object value) {
+      parameters.put(name, value);
+      return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Builder configName(String configName) {
+      this.configRef = configName;
+      return this;
+    }
+
+    /**
+     * Creates an Object of {@code type} and sets it's fields using an {@link OperationParameters} instance.
+     *
+     * @param name   the name of the operation parameter.
+     * @param type   the {@link Class} of the operation parameter to be instantiated.
+     * @param fields an {@link OperationParametersBuilder} instance with all the fields to be set for the object that is being created.
+     * @return {@code this} builder.
+     */
+    public Builder addParameter(String name, Class type, OperationParameters fields) {
+      this.complexParams.put(name, new Pair<>(type, fields));
```

Comment:
@marianogonzalez I don't want to expose this method in the public Interface, if any extension wants to extend from it, I will not make any sense to implement it.